blog.md

```diff
@@ -0,0 +1,94 @@
+
+![Alt Text](https://i.imgur.com/XQu2tug.png)
+
+For this dataset, I want to only extract the player, the season, their team, and their points per game. 
+Going back to our *td* pairs, I can click inspect element on each element and see what the corresponding *data-row=* is for each statistic. In HTML, they are tagged as 'player', 'season', 'team_id', 'pts_per_g'. We can create two Pandas dataframes to store our information: a temporary DataFrame we use to append a new page of information to our final set, and a final set to hold all the information.
+
+### Access Server Information
+
+At the start of each loop iteration, we are going to have to request an HTTP response for ourl url and store it as plain text. We then will pass it into a BeautifulSoup object, with the second parameter being a built in parser, cleverly named 'html.parser'. Our BeautifulSoup parser has two related functions to call: `find()` and `findAll()`. We can use `find()` to find the instance of our table. This is as opposed to using `findAll()`, when we need to search the entire document for instances of something, find returns the singular result we want directly rather than a list. For us, we are looking for up to 100 instances of relevant rows per page, so we want to use `findAll()`. We can create two for loops: one to loop over each relevant feature we want (player, season, etc.), and another one to loop over each *td* pair in our `findAll(*td*, *data-stat=feature*)`. Since we want the information as plain text, we have to convert it concurrently.
+
+![Alt Text](https://i.imgur.com/l5MaQso.png)
+
+# Append Data
+
+We now have our data, we want to re-format our data and add it to a final set. We take the stats variable and pass it into a Pandas DataFrame. Due to how the information was stored, our stats variable is a 4 element list with each element being size 100. We want the opposite of that, so we call `.transpose()` on our new dataframe to flip the rows to columns and vice versa, and set the column names to our list of features. We then call `.append()` with the final set DataFrame.
+
+# Bonus: Cleaning The Data
+
+We were able to work with a cleaned data set that didn't require too much to change. However, one may want to make more specific queries on the existing data set or eliminate any potential null values. For example, I can use the `.isin()` command to extract a value or list of values. So, if I only wanted the list of twenty point per game scorers from the Suns, Bulls, Lakers, Celtics and Knicks, I can extract them with `isin()`. For null values, imagine an older data set that started from 1964 instead of 1984. The three point line didn't exist before 1979, so I can use `dropna()` to remove any instances of seasons before the three point era. This snippet below showcases a basic use of `isin()` and `dropna()`. 
```

Comment:
```suggestion
We were able to work with a cleaned data set that didn't require too much to change. However, one may want to make more specific queries on the existing data set or eliminate any potentially null or missing values. For example, I can use the `.isin()` command to extract a value or list of values. So, if I only wanted the list of twenty point per game scorers from the Suns, Bulls, Lakers, Celtics and Knicks, I can extract them with `isin()`. For null values, imagine an older data set that started from 1964 instead of 1984. The three point line didn't exist before 1979, so I can use `dropna()` to remove any instances of seasons before the three point era. This snippet below showcases a basic use of `isin()` and `dropna()`. 
```

adding "or missing" to the first use of null, hopefully helps things be less ambiguous